src/test/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProviderTest.java

```diff
@@ -0,0 +1,279 @@
+
+    private static final List<String> validStdout = List.of("newPartition|worker1|1|ALL");
+    private static final List<String> validListOut = List.of("normal|worker1|1|ALL", "partition1||worker[21-2]|admin",
+            "partition2|worker2|1|slurm");
+    private static final List<String> validStdoutUpdatedPartition = List.of("updatingPartitionName|worker1,worker2|"
+            + "2|ALL");
+    private static final String[] CREATE_COMMAND_PATTERN = new String[]{"scontrol CREATE PartitionName=newPartition"
+            + "AllowGroups=ALL Nodes=worker1"};
+    private static final String[] LIST_COMMAND_PATTERN = new String[]{"sinfo -h -o \"%R|%N|%c|%g\""};
+    private static final String[] DELETE_COMMAND_PATTERN = new String[]{"scontrol DELETE "
+            + "PartitionName=deletedPartiiton"};
+    private static final String[] UPDATE_COMMAND_PATTERN = new String[]{"scontrol UPDATE "
+            + "PartitionName=updatingPartitionName AllowGroups=ALL Nodes=worker1,worker2"};
+    private static final String[] SINFO_COMMAND_PATTERN = new String[]{"sinfo --partition=updatingPartitionName "
+            + " -h -o \"%R|%N|%c|%g\""};
+    private static final String SCONTROL_COMMAND_TEMPLATE = "scontrolCommand";
+    private static final String SINFO_COMMAND_TEMPLATE = "sinfo";
+    private static final List<String> ownerList = List.of("root", "slurm");
+    private static final List<String> parallelEnvList = List.of("someVar1", "someVar1");
+    private static final List<String> nodeListOneNode = List.of("worker1");
+    private static final List<String> nodeListTwoNodes = new ArrayList<>(List.of("worker1", "worker2"));
+    private static final List<String> groupsList = new ArrayList<>(List.of("ALL"));
+
+    private final Map<String, Integer> slotDescriptionOneNode = Map.of("worker1", 1);
+
+    @Autowired
+    private SlurmQueueProvider slurmQueueProvider;
+
+    @MockBean
+    private SimpleCmdExecutor mockCmdExecutor;
+
+    @MockBean
+    private GridEngineCommandCompiler mockCommandCompiler;
+
+    @ParameterizedTest
+    @MethodSource("incorrectDataForPartitionCreateAndUpdate")
+    public void registerAndUpdatePartitionIncorrectDataException(final QueueVO incorrectRequest) {
+        final GridEngineException gridExceptionCreate = Assertions.assertThrows(GridEngineException.class,
+                () -> slurmQueueProvider.registerQueue(incorrectRequest));
+        Assertions.assertNotNull(gridExceptionCreate.getMessage());
+
+        final GridEngineException gridExceptionUpdate = Assertions.assertThrows(GridEngineException.class,
+                () -> slurmQueueProvider.updateQueue(incorrectRequest));
+        Assertions.assertNotNull(gridExceptionUpdate.getMessage());
+    }
+
+    @Test
+    public void registerPartitionSuccessful() {
+        final CommandResult creationResult = CommandResult.builder()
+                .stdOut(validStdout)
+                .stdErr(Collections.emptyList())
+                .exitCode(0).build();
```

Comment:
we could not specify `exitCode` value as `0` - it is the default `int` value
the same in several places

Reply:
removed